src/core/annotation.js

```diff
@@ -380,6 +362,42 @@ var Annotation = (function AnnotationClosure() {
     },
 
     /**
+     * Set the (normal) appearance.
+     *
+     * @public
+     * @memberof Annotation
+     * @param {Dict} dict - The annotation's data dictionary
+     */
+    setAppearance: function Annotation_setAppearance(dict) {
+      this.appearance = null;
+
+      var appearanceStates = dict.get('AP');
+      if (!isDict(appearanceStates)) {
+        return;
+      }
+
+      var normalAppearanceState = appearanceStates.get('N');
+      if (!isDict(normalAppearanceState) &&
+          !isStream(normalAppearanceState)) {
+        return;
+      }
+
+      // In case the normal appearance is a stream, then it is used directly.
+      if (isStream(normalAppearanceState)) {
+        this.appearance = normalAppearanceState;
+        return;
+      }
```

Comment:
I'm thinking that you may be able to simplify the conditions above slightly.
Wouldn't the following code work just as well?
```javascript
// In case the normal appearance is a stream, then it is used directly.
if (isStream(normalAppearanceState)) {
  this.appearance = normalAppearanceState;
  return;
}
if (!isDict(normalAppearanceState)) {
  return;
}
// In case the normal appearance is a dictionary, the `AS` entry provides
// the key of the stream in this dictionary.
...
```